src/main/java/com/NumNums/controlllers/LoginController.java

```diff
@@ -4,39 +4,40 @@
 import com.NumNums.models.User;
 import com.NumNums.models.service.UserService;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Controller;
-import org.springframework.ui.Model;
-import org.springframework.validation.Errors;
-import org.springframework.web.bind.annotation.ModelAttribute;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.ModelAndView;
 
-import javax.validation.Valid;
 
-@Controller
-public class LoginController {
+    @Controller
+    public class LoginController {
 
-    @Autowired
-    private UserService userService;
-
-    @RequestMapping(value= "NumNums/login", method = RequestMethod.GET)
-    public String displayLoginForm(Model model) {
-        model.addAttribute("title", "Log In");
-        return "login/login";
-    }
-
-    @RequestMapping(value = "NumNums/login", method = RequestMethod.POST)
-    public String processLoginForm(Model model) {
+        @Autowired
+        private UserService userService;
 
+        @RequestMapping(value = "NumNums/login", method = RequestMethod.GET)
+        public ModelAndView login() {
+            ModelAndView modelAndView = new ModelAndView();
+            Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+            User user = userService.findUserByEmail(auth.getName());
+            if (user != null) {
+                modelAndView.addObject("user", "Welcome,  " + user.getUsername());
+                modelAndView.addObject("adminMessage", "Content Available Only for Users with Admin Role");
+                modelAndView.setViewName("admin/home");
+            } else {
 
+                modelAndView.addObject("title", "Log In");
```

Comment:
Cool, I wonder if here would be a good time to let them know why they werent able to login (errors?)

Also, at this point you only have `ADMINS` right?